v4l2rtspserver-master/src/ImpEncoder.cpp

```diff
@@ -416,7 +416,9 @@ static void *update_thread(void *p) {
             if (ret < 0) {
                 LOG_S(INFO) << "Unable to get param to change the bitrate";
             }
+            attr.attrH264Smart.maxBitRate = (uint)newConfig->bitrate;
             attr.attrH264Vbr.maxBitRate = (uint)newConfig->bitrate;
+            attr.attrH264Cbr.outBitRate = (uint)newConfig->bitrate;
             IMP_Encoder_SetChnAttrRcMode(0, &attr);
```

Comment:
Thanks, I did wonder about that, but saw no errors, so thought it might be ok. But if it's setting random things in mem, that's not good! Thanks. 

I've pushed a fix to check the mode in use first.